src/RA_Condition.h

```diff
@@ -60,46 +74,75 @@ class CompVariable
     };
 
 public:
-    void Set(ComparisonVariableSize nSize, Type nType, unsigned int nInitialValue)
+    inline constexpr CompVariable() noexcept = default;
+    explicit inline constexpr CompVariable(_In_ MemSize nSize,
+                          _In_ CompVariable::Type nType,
+                          _In_ unsigned int nInitialValue) noexcept :
+        m_nVarSize{ nSize },
+        m_nVarType{ nType },
+        m_nVal{ nInitialValue }
+    {
+
+    }
+
+    explicit inline constexpr CompVariable(_In_ unsigned int nInitialValue,
+                                           _In_ unsigned int nPrevVal) noexcept :
+        m_nVal{ nInitialValue },
+        m_nPreviousVal{ nPrevVal }
     {
-        m_nVarSize = nSize;
-        m_nVarType = nType;
-        m_nVal = nInitialValue;
     }
 
-    void SetValues(unsigned int nVal, unsigned int nPrevVal)
+    _CONSTANT_FN Set(_In_ MemSize nSize,
+                     _In_ CompVariable::Type nType,
+                     _In_ unsigned int nInitialValue) noexcept
     {
-        m_nVal = nVal;
-        m_nPreviousVal = nPrevVal;
+        this->swap(CompVariable{ nSize, nType, nInitialValue });
     }
 
-    void ResetDelta()
+    _CONSTANT_FN SetValues(_In_ unsigned int nVal, _In_ unsigned int nPrevVal) noexcept
     {
-        m_nPreviousVal = m_nVal;
+        this->swap(CompVariable{ nVal, nPrevVal });
```

Comment:
`swap` might work for `Set` because previous value is unimportant when `Set` is called, but this will not work here. The size and type variables must not be overridden.

Reply:
I got you. I'll get rid of swap while I'm at it.